sync2/multipeer/multipeer.go

```diff
@@ -379,7 +403,9 @@ func (mpr *MultiPeerReconciler) Run(ctx context.Context) error {
 	lastWasSplit := false
 LOOP:
 	for {
-		interval := mpr.cfg.SyncInterval
+		interval := time.Duration(
+			float64(mpr.cfg.SyncInterval) *
+				(1 + mpr.cfg.SyncIntervalSpread*(rand.Float64()*2-1)))
```

Comment:
I'm a bit confused by this interval calculation. Would it be simpler if `SyncIntervalSpread` would be defined as `time.Duration` and gave the maximum deviation from the interval?

```go
interval := mpr.cfg.SyncInterval + rand.N(mpr.cfg.SyncIntervalSpread)
```

This will uniformly generate a duration between `[SyncInterval, SyncInterval+SyncIntervalSprad)` while the current definition is `[SyncInterval, SyncInterval+2*SyncIntervalSpread)` which is a bit odd to me?

Reply:
The idea was for `SyncIntervalSpread` to be a floating point number `0..1` and to have intervals between `[SyncInterval - SyncInterval*SyncIntervalSpread, SyncInterval + SyncInterval*SyncIntervalSpread]`
We could of course also use `MinSyncInterval` and `MaxSyncInterval`, but I'm not sure which is more convenient.
My idea was that if I e.g. want the actual sync interval to be uniformly spread across `SyncInterval +/- 25%` I just set `SyncIntervalSpread` to `0.25`

Reply:
Maybe I should reflect this simpler explanation in the comments, incl. godoc comments for the config struct

Reply:
I would prefer the `Min-` and `Max-` config options, but if you think a fractional spread is easier then go for that. But please add some explanation - to the config and/or here - what the values mean 🙂